tests/unit/transports/socket_transport_test.py

```diff
@@ -25,51 +25,79 @@ def test_init(self):
-        expect(sock.connect).args(('host.net', 5309))
         expect(sock.settimeout).args(None)
 
+    def test_connect_with_no_klass_arg_ipv4(self):
+        sock = mock()
+        expect(socket.socket).returns(sock)
+        self._set_up_connect_test(sock)
+        expect(socket, 'getaddrinfo').args(
+            'host', 5309, 0, 0, socket.IPPROTO_TCP,
+        ).returns(
+            [(socket.AF_INET, socket.SOCK_STREAM, 0, 'canon', ('host.net', 5309))]
+        )
+        expect(sock.connect).args(('host.net', 5309))
+        self.transport.connect(('host', 5309))
+
+    def test_connect_with_no_klass_arg_ipv6(self):
+        sock = mock()
+        expect(socket.socket).returns(sock)
+        self._set_up_connect_test(sock)
+        expect(socket, 'getaddrinfo').args(
+            'host', 5309, 0, 0, socket.IPPROTO_TCP,
+        ).returns(
+            [
+                (
+                    socket.AF_INET6, socket.SOCK_STREAM, 0, 'canon',
+                    ('::1', 5309, 0, 0),
+                ),
+            ]
+        )
+        expect(sock.connect).args(('::1', 5309, 0, 0))
         self.transport.connect(('host', 5309))
 
     def test_connect_with_klass_arg_ipv4(self):
         klass = mock()
         sock = mock()
         expect(klass).returns(sock)
-
-        self.connection._connect_timeout = 4.12
-        self.connection._sock_opts = {
-            ('family', 'tcp'): 34,
-            ('range', 'ipv6'): 'hex'
-        }
-
-        expect(sock.setblocking).args(True)
-        expect(sock.settimeout).args(4.12)
-        expect(sock.setsockopt).any_order().args(
-            'family', 'tcp', 34).any_order()
-        expect(sock.setsockopt).any_order().args(
-            'range', 'ipv6', 'hex').any_order()
-        expect(socket,'getaddrinfo').args('host',5309,0,0,socket.IPPROTO_TCP).returns(
-            [('family','socktype','proto', 'canon',('host.net',5309))] )
+        self._set_up_connect_test(sock)
+        expect(socket, 'getaddrinfo').args(
+            'host', 5309, 0, 0, socket.IPPROTO_TCP,
+        ).returns(
+            [(socket.AF_INET, socket.SOCK_STREAM, 0, 'canon', ('host.net', 5309))]
+        )
         expect(sock.connect).args(('host.net', 5309))
-        expect(sock.settimeout).args(None)
+        self.transport.connect(('host', 5309), klass=klass)
 
+    def test_connect_with_klass_arg_ipv6(self):
+        klass = mock()
+        sock = mock()
+        expect(klass).returns(sock)
+        self._set_up_connect_test(sock)
+        expect(socket, 'getaddrinfo').args(
+            'host', 5309, 0, 0, socket.IPPROTO_TCP,
+        ).returns(
+            [
+                (
+                    socket.AF_INET6, socket.SOCK_STREAM, 0, 'canon',
+                    ('::1', 5309, 0, 0),
+                ),
+            ]
+        )
+        expect(sock.connect).args(('::1', 5309, 0, 0))
         self.transport.connect(('host', 5309), klass=klass)
 
```

Comment:
Missing test coverage for the following code paths that are not exercised by existing tests:

1. `socket.getaddrinfo` fails. It normally raises `socket.gaierror`
2. All connection attempts failed
3. First connection attempt failed, but a subsequent one succeeded.